src/components/sidebar/sidebar.component.js

```diff
@@ -11,76 +11,79 @@ import Box from "../box";
+const Sidebar = React.forwardRef(
+  (
+    {
+      open,
+      disableEscKey,
+      enableBackgroundUI,
+      header,
+      position,
+      size,
+      children,
+      onCancel,
+      ...rest
+    },
+    sideBarRef
+  ) => {
+    const closeIcon = () => {
+      if (!onCancel) return null;
+      return (
+        <IconButton aria-label="close" onAction={onCancel} data-element="close">
+          <Icon type="close" />
+        </IconButton>
+      );
+    };
 
-  const closeIcon = () => {
-    if (!onCancel) return null;
-    return (
-      <IconButton aria-label="close" onAction={onCancel} data-element="close">
-        <Icon type="close" />
-      </IconButton>
-    );
-  };
+    const componentTags = {
+      "data-component": "sidebar",
+      "data-element": rest["data-element"],
+      "data-role": rest["data-role"],
+    };
 
-  const componentTags = {
-    "data-component": "sidebar",
-    "data-element": rest["data-element"],
-    "data-role": rest["data-role"],
-  };
-
-  const sidebar = (
-    <SidebarStyle
-      ref={sideBarRef}
-      position={position}
-      size={size}
-      data-element="sidebar"
-      role="complementary"
-      onCancel={onCancel}
-    >
-      {closeIcon()}
-      {header && <SidebarHeader>{header}</SidebarHeader>}
-      <Box
-        data-element="sidebar-content"
-        p={4}
-        pt="27px"
-        scrollVariant="light"
-        overflow="auto"
+    const sidebar = (
+      <SidebarStyle
+        ref={sideBarRef}
+        position={position}
+        size={size}
+        data-element="sidebar"
+        role="complementary"
+        onCancel={onCancel}
       >
-        <SidebarContext.Provider value={{ isInSidebar: true }}>
-          {children}
-        </SidebarContext.Provider>
-      </Box>
-    </SidebarStyle>
-  );
+        {closeIcon()}
+        {header && <SidebarHeader>{header}</SidebarHeader>}
+        <Box
+          data-element="sidebar-content"
+          p={4}
+          pt="27px"
+          scrollVariant="light"
+          overflow="auto"
+        >
+          <SidebarContext.Provider value={{ isInSidebar: true }}>
+            {children}
+          </SidebarContext.Provider>
+        </Box>
+      </SidebarStyle>
+    );
 
-  return (
-    <Modal
-      open={open}
-      onCancel={onCancel}
-      disableEscKey={disableEscKey}
-      enableBackgroundUI={enableBackgroundUI}
-      className="carbon-sidebar"
-      {...componentTags}
-    >
-      {enableBackgroundUI ? (
-        sidebar
-      ) : (
-        <FocusTrap wrapperRef={sideBarRef}>{sidebar}</FocusTrap>
-      )}
-    </Modal>
-  );
-};
+    return (
+      <Modal
+        open={open}
+        onCancel={onCancel}
+        disableEscKey={disableEscKey}
+        enableBackgroundUI={enableBackgroundUI}
+        className="carbon-sidebar"
+        {...componentTags}
+      >
+        {enableBackgroundUI ? (
+          sidebar
+        ) : (
+          <FocusTrap wrapperRef={sideBarRef}>{sidebar}</FocusTrap>
```

Comment:
We need to handle for scenarios where the ref isn't passed in by the implementing project as well or this will be a breaking change: the focus trap will require a ref to to focus the first element.

```
const sidebarRef = ref || useRef()
```